frontend/src/components/ingress/Details.tsx

```diff
@@ -42,6 +43,110 @@ export default function IngressDetails() {
     );
   }
 
+  /** this function is used to determine the protocol used in the formatURL for LinkedStringFormat */
+  function getProtocol(item: any, url: any) {
+    const hostList: string[] = item.jsonData.spec?.tls?.map(({ ...hosts }) => `${hosts.hosts}`);
+    const isHttps = hostList.includes(`${url}`);
+
+    return isHttps;
+  }
+
+  /** this function is used to format the url to be used in the Link component */
+  function LinkStringFormat(url: any, item?: any, urlPath?: any) {
+    let urlProtocol;
+
+    if ((url === '*' || url === '' || url === undefined) && !urlPath) {
+      return <Typography>{url}</Typography>;
+    }
+
+    if ((url === '*' || url === '' || url === undefined) && urlPath) {
+      return <Typography>{urlPath}</Typography>;
+    } else {
+      /* if the ingress does not use tls, then the url should be http */
+      if (!item.jsonData.spec?.tls) {
+        urlProtocol = 'http://';
+      } else {
+        /* if the ingress does use tls, then we compare the url to the hosts in the tls array to determine if the url should be http or https */
+        getProtocol(item, url) ? (urlProtocol = 'https://') : (urlProtocol = 'http://');
+      }
+
+      /* since we cannot access the prefix from the ingress object, we have to access it from the itemData array */
+      const itemData: any[] = item.jsonData.spec?.rules;
```

Comment:
You can actually use `item.spec.rules`. Because there's a spec getter in the class.
And the itemDatas should be called rules IMO.